main.py

```
@@ -148,213 +91,234 @@ async def send_message(self, connection_id: str, message: str):
+
+@app.delete("/user/api-keys/{key_id}")
+async def delete_user_api_key(key_id: int, current_user: models.User = Depends(auth.get_current_active_user), db: Session = Depends(get_db)):
+    key = db.query(models.APIKey).filter(models.APIKey.id == key_id, models.APIKey.user_id == current_user.id).first()
+    if not key:
+        raise HTTPException(status_code=404, detail="API Key not found")
+    db.delete(key)
+    db.commit()
+    return {"message": "API Key deleted"}
+
+@app.get("/user/stats")
+async def get_user_stats(current_user: models.User = Depends(auth.get_current_active_user), db: Session = Depends(get_db)):
```

Ruff (0.15.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)